src/zope/sqlalchemy/datamanager.py

```diff
@@ -317,14 +306,13 @@ def after_bulk_delete(self, delete_context):
                      self.transaction_manager, self.keep_session)
 
     def before_commit(self, session):
-        # Support SQLAlchemy 2.0
-        transaction = (
-            session.get_transaction()
-            if hasattr(session.transaction, "get_transaction")
-            else session.transaction
+        in_nested_transaction = (
+            session.in_nested_transaction()
+            if sqlalchemy_version >= "1.4.0"
+            else session.transaction.nested
```

Comment:
A comment here referencing version 1.3 would be fine, so we can find this line of code once we drop support for versions older than 1.4.